audio.c

```diff
@@ -1293,7 +1294,7 @@ void cRpiAudioDecoder::Reset(void)
 	Lock();
 	m_reset = true;
 	m_wait->Signal();
-	while (m_reset)
+	while (m_reset && !m_terminated)
```

Comment:
Why not checking for cThread::Running() when entering the method?

Reply:
I will try that once more. I think that I tried it and it did not prevent a hang.

Reply:
You seem to be right, checking for `Running()` seems to suffice. Maybe last time I tried it, I had simply hit the hang in `cOmx::DeInit()`.

Reply:
Actually, checking for `Running()` is wrong, because it is only a request for the `Action()` to terminate. 05bf186eb1606c741068737c7b3f49e8032d21b4 checks for `cThread::Active()`.